packages/preview-middleware-client/src/flp/bootstrap.js

```diff
@@ -13,9 +11,9 @@ async function ushellBootstrap(fnCallback) {
     try {
         const response = await fetch('/resources/sap-ui-version.json');
         const json = await response.json();
-        const version = json?.libraries?.find((lib) => lib.name === 'sap.ui.core')?.version;
-        const major = version ? parseInt(version.split('.')[0], 10) : 2;
-        if (major >= 2) {
+        const version = json?.libraries?.find((lib) => lib.name === 'sap.ui.core')?.version ?? '1.121.0';
+        const [ majorUi5Version ] = version.replace(/snapshot-untested|snapshot-|snapshot/, '').split('.').map(Number);
```

Comment:
I don't think we should limit what kind of suffixes the version can have, because if UI5 decides to add a new one it will break this logic. Currently we do not use the minor version, so corrupting that part would not cause an issue right now. 

Also `parseInt` seems to be better suited here, because it will only take the first numeric characters and ignore everything after the first non numeric character. Which means we do not need to do replacement then either.

Reply:
I basically copied that suffix handling from the adp/ui5-version-utils. But you are right. I adjusted it.